pyomo/contrib/solver/solvers/sol_reader.py

```diff
@@ -10,28 +10,153 @@
 #  ___________________________________________________________________________
 
 
```

Comment:
Should this file go in the common directory?

Reply:
As of right now, no, because it is solver-specific. It can be moved later.